### PR TITLE
fix: dev: #113: Weight for disk space used should consider the concurrency level of guava cache

### PR DESCRIPTION
**Pull request by abhishekdas99**

### CODE REVIEW

**Comment by vrajat** [comment]
Can you provide any details on manual testing you have done to ensure that this solves the cache eviction problem ?

**Comment by abhishekdas99** [comment]
@stagraqubole @vrajat  Can anyone take a look at the PR ?

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java [hunk 1]
```diff
@@ -105,6 +106,7 @@
   private static final int DEFAULT_SERVER_MAX_THREADS = Integer.MAX_VALUE;
   private static final int DEFAULT_SERVER_PORT = 8899;
   private static final int DEFAULT_SOCKET_READ_TIMEOUT = 30000; // ms
+  private static final int DEFAULT_CACHE_CONCURRENCY_LEVEL = 4;
```
Comment (vrajat) [inline rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: A way to turn off the change is required. Can the default value be 1 ? That way the change is not _rolled out_. Based on an account feature we can set it to 4. For any other value, a bootstrap can be used.

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java [hunk 2]
```diff
@@ -290,6 +292,16 @@ public static boolean isParallelWarmupEnabled(Configuration conf)
     return conf.getBoolean(KEY_PARALLEL_WARMUP, DEFAULT_PARALLEL_WARMUP);
   }
 
+  public static int getCacheConcurrencyLevel(Configuration conf)
```
Comment (vrajat) [inline rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: Is a unit test required ? I think @jordanw-bq added unit tests for this file.

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java (hunk 1 above)
Comment (abhishekdas99) [reply rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: The value has been 4. We are not changing the default value. We just changed the weight for the dummy key.
The default value can be found here.
https://github.com/google/guava/blob/master/guava/src/com/google/common/cache/CacheBuilder.java#L159

Comment (vrajat) [reply rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: Hmm. I am out of my depth here. I interpreted the change as:
- Before _getWeight_ function returned 100 but now returns 25
- There maybe unintended consequences due to the change in weight for the same _usedMb_. 
- If that is true, then a roll out is requored.

**Comment by abhishekdas99** [comment]
This PR is obsolete because of this commit 61bba2afdf5eb300acce6051bcf3420e2e8472de. We have got rid off the DiskMonitorService

**Pull request closed without merging**